packages/transaction-controller/src/TransactionController.ts

```diff
@@ -521,16 +530,37 @@ export class TransactionController extends BaseControllerV1<
     this.publish =
       hooks?.publish ?? (() => Promise.resolve({ transactionHash: undefined }));
 
+    this.onNetworkStateChange = onNetworkStateChange;
+    this.isMultichainEnabled = isMultichainEnabled;
+    this.incomingTransactions = incomingTransactions;
+    this.getNetworkClientRegistry = getNetworkClientRegistry;
+  }
+
+  /**
+   * Initializes `registry`, `nonceTracker`, `#multichainTrackingHelper`,
+   * `incomingTransactionHelper`, `pendingTransactionTracker`, and passes a
+   * callback to `onNetworkStateChange`.
+   *
+   * This relies on the provider and block tracker being defined which may not
+   * have been the case at the time the Transaction Controller was instantiated.
+   *
+   * @param passedProvider - Reference to the provider proxy object
+   * @param passedBlockTracker - Reference to the block tracker proxy object
+   */
+  delayedInit(passedProvider: Provider, passedBlockTracker: BlockTracker) {
```

Comment:
Why is this called `delayedInit`? Is there a non-delayed initialization? This should be the only initialization step